deepchecks/vision/metrics_utils/semantic_segmentation_metrics.py

```diff
@@ -0,0 +1,60 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module for calculating semantic segmentation metrics."""
+from collections import defaultdict
+from typing import Tuple
+
+import numpy as np
+import torch
+from ignite.metrics import Metric
+
+
+class MeanDice(Metric):
+    """Metric that calculates the mean Dice metric for each class.
+
+    See more: https://en.wikipedia.org/wiki/S%C3%B8rensen%E2%80%93Dice_coefficient
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._evals = defaultdict(lambda: {'dice': 0, 'count': 0})
+
+    def reset(self) -> None:
+        """Reset metric state."""
+        super().reset()
+        self._evals = defaultdict(lambda: {'dice': 0, 'count': 0})
+
+    def update(self, output: Tuple[torch.Tensor, torch.Tensor]):
+        """Update metric with batch of samples."""
+        y_pred, y = output
+
+        for i in range(len(y)):
+            for class_id in [int(x) for x in torch.unique(y[i])]:
+                y_pred_this_label = (y_pred[i].argmax(0) == class_id).numpy()
+                y_this_label = (y[i] == class_id).numpy()
+
+                tp = np.logical_and(y_pred_this_label, y_this_label).sum()
+                total_y_pred = y_pred_this_label.sum()
+                total_y = y_this_label.sum()
```

Comment:
Since the sum is on bool, I'd call it y_counts and y_pred_counts rather then "total", but not critical

Reply:
:+1: